package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -0,0 +1,200 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+Dihedral and Ramachandran analysis --- :mod:`MDAnalysis.analysis.dihedrals`
+===========================================================================
+
+:Author: Henry Mull
+:Year: 2018
+:Copyright: GNU Public License v2
+
+.. versionadded:: 0.18.1
+
+This module calculates the dihedral angles phi and psi in degrees for a given
+list of residues (or atoms corresponding to the residues). This can be done for
+selected frames or whole trajectories.
+
+A list of time steps that contain phi and psi angles for each residue is
+generated, and a basic Ramachandran plot can be generated using the attribute
+:meth:`Ramachandran.plot`. This plot is best used as a reference, but it also
+allows for user customization.
+
+
+See Also
+--------
+:mod:`MDAnalysis.lib.distances.calc_diherals`
+
+
+Example application
+-------------------
+This example will show how to calculate the phi and psi angles of adenylate
+kinase and generate a basic Ramachandran plot. The trajectory is included with
+the test data files::
+
+   import MDAnalysis as mda
+   from MDAnalysisTests.datafiles import GRO, XTC
+   u = mda.Universe(GRO, XTC)
+   r = u.select_atoms("protein")    # selection of residues
+
+   from MDAnalysis.analysis.dihedrals import Ramachandran
+   R = Ramachandran(r)
+   R.run()
```

Comment:
Preferred is `R = Ramachandran(r).run()` as a one-liner. The `run` method returns the object itself.

Reply:
That's originally what I was doing but I've been using `rms.py` as a reference so I thought that was the convention. I'll be sure to change it back.

Reply:
This might still be in older modules. We changed the return value of run only after the analysis class was introduced.